package sglib.publicproto;

// 游戏信息
message GameInfo 
{
	required int32 gameid		= 1;
	required int32 cur_count	= 2;
}

// 单个游戏服务器的信息
message GameServerInfo
{
	required int32 serverid		= 1;
	required int32 cur_count	= 2;
}

// 单个gate服务器的信息
message GateServerInfo
{
	required int32 id		= 1;
	required string ip		= 2;
	required int32 port		= 3;
	required int32 cur_count= 4;
	required int32 max_count= 5;
}

message RoomPlayerInfo
{
	required string user	= 1;
	required int32 isready	= 2;
}

message GroupInfo
{
	required uint64 id		= 1;	// 群id
	required string name	= 2;	// 群名字
	required string icon	= 3;	// 群图标
	required int32 count	= 4;	// 群总人数
}

message GroupMemberInfo
{
	required string user		= 1;	// 帐号
	required string nickname	= 2;	// 昵称
	required string head		= 3;	// 头像
	required int32  online		= 4;	// 是否在线 1:在线 0:离线
}

message GroupMemberPosInfo
{
	required string user		= 1;	// 帐号
	required int32 gateresid	= 2;	// 玩家所在的group gate 的 res id(xml中配置的id)
	required uint64 clientid	= 3;	// 玩家在此gate上的 id 
}

